experimental/listener.go

```diff
@@ -37,7 +37,7 @@ type FunctionListener interface {
 	//   - paramValues:  api.ValueType encoded parameters.
 	//
 	// Note: api.Memory is meant for inspection, not modification.
-	Before(ctx context.Context, mod api.Module, def api.FunctionDefinition, paramValues []uint64) context.Context
+	Before(ctx context.Context, mod api.Module, def api.FunctionDefinition, paramValues []uint64, stackIterator api.StackIterator) context.Context
 
 	// After is invoked after a function is called.
```

Comment:
I only provided the `StackIterator` to the `Before` function because we only need it there at the moment, and it feels like it can be easily stored ahead of time in `Before`, only to be retrieved in the subsequent `After` call. It would maybe make sense to add there as well before the need arises. Let me know.

Reply:
we can wait until then, but agree.